architecture/2025-04-compute-extensibility.md

```diff
@@ -0,0 +1,469 @@
+arguments/commands, or Go APIs for shared components should provide this
+section. Write N/A here if not applicable.
+- Describe the REST APIs in detail for new resource types or updates to
+  existing resource types. E.g. API Path and Sample request and response.
+- Describe new commands in the CLI or changes to existing CLI commands.
+- Describe the new or modified Go APIs for any shared components.
+-->
+
+<!--
+### CLI Design (if applicable)
+Include if applicable – any design that changes Radius CLI
+arguments/commands. Write N/A here if not applicable.
+- Describe new commands in the CLI or changes to existing CLI commands.
+-->
+
+<!--
+### Implementation Details
+High level description of updates to each component. Provide information on 
+the specific sub-components that will be updated, for example, controller, processor, renderer,
+recipe engine, driver, to name a few.
+
+#### UCP (if applicable)
+#### Bicep (if applicable)
+#### Deployment Engine (if applicable)
+#### Core RP (if applicable)
+#### Portable Resources / Recipes RP (if applicable)
+-->
+
+<!--
+### Error Handling
+Describe the error scenarios that may occur and the corresponding recovery/error handling and user experience.
+-->
+
+<!--
+## Test plan
+
+Include the test plan to validate the features including the areas that
+need functional tests.
+
+Describe any functionality that will create new testing challenges:
+- New dependencies
+- External assets that tests need to access
+- Features that do I/O or change OS state and are thus hard to unit test
+-->
+
+<!--
+## Security
+
+Describe any changes to the existing security model of Radius or security 
+challenges of the features. For each challenge describe the security threat 
+and its mitigation with this design. 
+
+Examples include:
+- Authentication 
+- Storing secrets and credentials
+- Using cryptography
+
+If this feature has no new challenges or changes to the security model
+then describe how the feature will use existing security features of Radius.
+-->
+
+<!--
+## Compatibility (optional)
+
+Describe potential compatibility issues with other components, such as
+incompatibility with older CLIs, and include any breaking changes to
+behaviors or APIs.
+-->
+
+<!--
+## Monitoring and Logging
+
+Include the list of instrumentation such as metric, log, and trace to 
+diagnose this new feature. It also describes how to troubleshoot this feature
+with the instrumentation. 
+-->
+
+<!--
+## Development plan
+
+Describe how you will deliver your features. This includes aligning work items
+to features, scenarios, or requirements, defining what deliverable will be
+checked in at each point in the product and estimating the cost of each work
+item. Don’t forget to include the Unit Test and functional test in your
+estimates.
+-->
+
+## Open Questions
+
+* Should we convert Kubernetes deployments to recipes with this work, after this work, or not at all?
```

Comment:
I'm not sure I understand this question. You can already define Kubernetes deployments using recipes

Reply:
Yes this is vaguely worded. I'll add context to the question.